src/types/string.rs

```diff
@@ -227,6 +227,29 @@ impl PyString {
         }
     }
 
+    /// Returns an iterator over the PyString.
+    ///
+    /// Does not allocate anything (python or rust heap).
+    pub fn chars(&self) -> impl ExactSizeIterator<Item = PyResult<char>> + '_ {
+        unsafe {
+            let len = ffi::PyUnicode_GetLength(self.as_ptr());
+            (0..len).map(move |i| {
```

Comment:
Since the implementation is based on indexing, would it make sense to expose this a the interface as well? Something like `char_at(&self, index: usize) -> Option<char>` so that the iterator can be produced on the outside?